File: src/main/java/com/ject/studytrip/dummy/application/service/DummyStampCommandService.java
package com.ject.studytrip.dummy.application.service;

import com.ject.studytrip.dummy.application.dto.CreateDummyStampCommand;
import com.ject.studytrip.dummy.application.generator.CreateDummyStampCommandGenerator;
import com.ject.studytrip.stamp.domain.factory.StampFactory;
import com.ject.studytrip.stamp.domain.model.Stamp;
import com.ject.studytrip.trip.domain.model.Trip;
import org.springframework.stereotype.Service;

@Service
public class DummyStampCommandService {

    public Stamp createDummyStamp(Trip trip, int stampOrder) {
        CreateDummyStampCommand command =
                CreateDummyStampCommandGenerator.of(trip.getCategory(), stampOrder);

        return StampFactory.create(trip, command.name(), command.stampOrder(), command.endDate());
    }
}

File: src/main/kotlin/com/ject/studytrip/dummy/application/service/DummyStampCommandService.kt
package com.ject.studytrip.dummy.application.service

import com.ject.studytrip.dummy.application.dto.CreateDummyStampCommand
import com.ject.studytrip.stamp.domain.factory.StampFactory
import com.ject.studytrip.stamp.domain.model.Stamp
import com.ject.studytrip.trip.domain.model.Trip
import com.ject.studytrip.trip.domain.model.TripCategory
import org.springframework.stereotype.Service
import java.time.LocalDate

@Service
class DummyStampCommandService {
    fun createDummyStamp(
        trip: Trip,
        stampOrder: Int,
    ): Stamp {
        val command =
            when (trip.category) {
                TripCategory.COURSE -> CreateDummyStampCommand.of("testStamp", stampOrder, LocalDate.now().plusDays(10))
                TripCategory.EXPLORE -> CreateDummyStampCommand.of("testStamp", 0, null)
            }

        return StampFactory.create(trip, command.name, command.stampOrder, command.endDate)
    }
}
